### ISSUE TITLE
Images upload duplicate copies of each and fail to load in CP

### ISSUE THREAD

**Issue by jpgdcl** [statamic/v2-hub#704]
### Steps to reproduce
1. Create an entry with a replicator > asset field (herein called `gallery`)
2. Upload images to the `gallery`
3. Select these images and hit 'Select'
4. Watch as they duplicate in the bucket

### Expected behaviour
One copy of each image should appear in the asset upload

### Actual behaviour
Two copies of each image appear, one with a random numerical ID appended to it e.g. `chicken.jpg` and `chicken-1464036576.jpg` - also visible in the assets folder as files, alongside the normal one. When I try and delete the extra one both are deleted. Two sets of images are added to the frontend too.

After saving, in the CP, I get an infinite loading spinner. Tried clearing cache, to no avail.

Tried erasing the second set of IDs from the list of `images: ` in the markdown. Only one set now load in the frontend, but still hanging in the CP - sad face time :(

### Server configuration
**Operating system**: OSX 10.11

**Web server:** Apache (MAMP)

**PHP version:** 5.6.1.0

**Statamic version:** 2.0.5

**Updated from an older Statamic or fresh install:** Manual Ghetto Upgrizzle

**List of installed addons:** Useragent (not related)

#### Web server error log
```
MAMP PHP Log
[23-May-2016 20:50:39 UTC] PHP Fatal error:  Call to a member function isImage() on null in example.dev/statamic/core/Http/Controllers/AssetsController.php on line 93

```

#### Statamic error log (`/local/storage/logs/statamic.log`)
```
[2016-05-23 20:50:39] dev.ERROR: exception 'Symfony\Component\Debug\Exception\FatalErrorException' with message 'Call to a member function isImage() on null' in /example.dev/statamic/core/Http/Controllers/AssetsController.php:93
Stack trace:
#0 {main}  
```
(Might not be associated, could have been related to this: https://github.com/statamic/v2-hub/issues/703)

#### Browser log
```
Failed to load resource: the server responded with a status of 500 (Internal Server Error)
undefined:1 Uncaught (in promise) XMLHttpRequest
```

### Screenshot

<img width="890" alt="cp image bug" src="https://cloud.githubusercontent.com/assets/9616783/15574757/206cf346-2347-11e6-8e86-ddba5325c104.png">

**Comment by jpgdcl** [comment]
Now finding that when I don't have an image uploaded for a replicator, it seems to fallback to another image (with the same field name) from another set within the replicator...

e.g. Below if I upload an image to `set_dual` in the CP, it now appears in `set_image` even though I haven't even selected an image for that set.

```
{{ elseif type == "set_dual" }}

<p>{{ text }}</p>

 {{ assets:image }}
  <img src="{{ url }}"/>
 {{ /assets:image }}

{{ elseif type == "set_image" }}

{{ if image }}
 {{ assets:image }}
  <img src="{{ url }}"/>
 {{ /assets:image }}
{{ /if }}

...
```

I feel like the issues might be related as generally images seem to be all over the place.

I've tried removing all the images and starting again, but sadly that didn't work.

**Comment by jpgdcl** [comment]
The images also appear sometimes to be tied together. So if I erase the one with the random ID on the end of it, it deletes the corresponding one.

**Comment by jpgdcl** [comment]
Images also seem to be randomly disappearing now too... Very strange.

**Comment by jpgdcl** [comment]
**Update:**

For anyone else with the same problem, I tried to delete them manually from the `assets` folder but I'm finding some of them are still in: `/storage/assets/.../folder.yaml` so having to manually delete them.

<img width="328" alt="phantom" src="https://cloud.githubusercontent.com/assets/9616783/15605460/f8ec96d4-23fd-11e6-8480-1c3bf8728ee5.png">

I know that Statamic appends a timestamp to images with the same filename, so its obviously thinking I'm uploading each image twice.

I find it doesn't seem to happen as much if I drag and drop them into here (rather than hitting browse and dragging and dropping them into the asset manager):

<img width="884" alt="drag" src="https://cloud.githubusercontent.com/assets/9616783/15605425/c4f264ee-23fd-11e6-85ae-f612724d084d.png">

### Potential Fix

A good fix I think would at the very least, if an image doesn't load in the edit entry view, to just show that it hasn't loaded and allow the user to remove it. So at least you can still load the page. If this was a client site it would be a total nightmare.

**Comment by jpgdcl** [comment]
### Update

Think I just figured out part of the problem: The asset manager doesn’t seem to be deleting assets from the folder, so obviously it then appends the time stamp as there’s already a file in there with the same name.

**Comment by jasonvarga** [comment]
Just to confirm...
You were opening the 'browse' dialog, and uploading into the modal?

I _think_ I may have fixed this for 2.0.6. I was noticing that when you drag onto the dialog, it also thinks you're dragging onto the field. Sounds like thats where a potential double up could have been.
